### DISCUSSION TITLE
definitionHover: sometimes no source code for Go

### DISCUSSION THREAD

**Discussion by vitaly-zdanevich** [neoclide/coc.nvim#4501] [Q&A] [answered]
Sometimes I see the code, but sometimes something like this:

![image](https://user-images.githubusercontent.com/3514015/216693288-af6fcf10-5ddf-41be-be59-2e530badce27.png)

Or for Go you can recommend another Vim plugin?

**Comment by jackielii** [comment, marked as answer]
This looks like the right behaviour.

When hovering over a function, `gopls` will show the docs. When hovering a struct, the full method set will be shown. See [gopls hoverKind](https://github.com/golang/tools/blob/master/gopls/doc/settings.md#documentation) and "Method information in hover" in [gopls v0.10.0 release notes](https://github.com/golang/tools/releases/tag/gopls%2Fv0.10.0)

**Reply by vitaly-zdanevich** [reply]
Can I see the source for functions? Do I need to alter some config for that?

**Reply by jackielii** [reply]
I don't think so, that's gopls's behaviour, you can check gopls's source code to find out for sure.

**Reply by vitaly-zdanevich** [reply]
Do you know good alternative of Coc for this?

**Reply by vitaly-zdanevich** [reply]
Or maybe you can implement it by go-to-definition - rendering it in the popup?

**Reply by jackielii** [reply]
I think you're probably using language server `gopls` directly instead of a plugin and it's not possible to get it just via configuration. The best way is to write your own plugin to get this behaviour. Or file an issue at `gopls`'s repo to see if they think it make sense.

**Reply by vitaly-zdanevich** [reply]
> Or file an issue at gopls's repo to see if they think it make sense

Like "Please support getting of a function source code"?

**Reply by vitaly-zdanevich** [reply]
@jackielii Currently for this popup I have this:

`nnoremap <silent> <leader>h :call CocActionAsync('definitionHover')<cr>`

but can I ran it on mouse hover?

**Reply by jackielii** [reply]
I don't know, but that's not a question for coc.nvim, more of a Neovim or whatever terminal emulator you're using kind of question. Better to start there to get what you want.

A word of advice, focus on the job, not just the tool if it already does the job.

**Reply by chemzqm** [reply]
> mouse hover?

Mouse hover normally exists with gvim only, not terminal.

**Reply by vitaly-zdanevich** [reply]
Mouse hover exists for me in Vim, for example
![image](https://github.com/neoclide/coc.nvim/assets/3514015/505c676f-4866-42ad-8c0d-f14559675a46)

**Reply by chemzqm** [reply]
The mouse hover seems to be a terminal feature.
